### ISSUE TITLE
[Sev2] New annotation: Services label is not programmatically associated with Metrics combo box control.

### ISSUE THREAD

**Issue by JonCole** [grafana/grafana#117288]
**Steps to reproduce**:
1. Create a new Dashboard
2. Navigate to "Add visualization" button and hit ENTER key. "Select data source" popup will appear.
3. Navigate to the Azure Monitor list item select it.
4. Now navigate to the tool bar and select back to dashboard control and select. Now select setting control in the tool bar.
5. Navigate to  Annotations tab and select new query  button.
6. Navigate to all control under Annotations tab and New annotation dialog.
7. Verify whether Services label is programmatically associated with Metrics combo box control or not.

**Actual Result**:
Services label is not programmatically associated with Metrics combo box control.
Observation: 
When the combo box receives keyboard focus, it expands automatically and focus moves directly to the first list item.
Screen reader announcement (NVDA / JAWS / Narrator):
“4 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu. Select options menu, list. Metrics, 1 of 4.”
The associated label “Services” is not announced with the combo box.

**Expected Result**
When focus moves to the combo box, screen readers (NVDA / JAWS / Narrator) should announce the associated label, for example:
“Services, Metrics, combo box, collapsed.”
The combo box should expand only after explicit user interaction (Enter, Space, or Arrow keys), and the label association should remain intact when the list is expanded.

**Relevant WCAG Criteria:** [#.#.# WCAG Criterion](link to https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0)
WCAG: [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
Section 508: [E207.2 WCAG Conformance](https://www.access-board.gov/ict/#E207.2), [E205.4 Accessibility Standard](https://www.access-board.gov/ict/#E205.4) and [E205.4.1 Word Substitution when Applying WCAG to Non-Web Documents](https://www.access-board.gov/ict/#E205.4.1), and [E207.2.1 Word Substitution when Applying WCAG to Non-Web Software](https://www.access-board.gov/ict/#e207-2-1-word-substitution-when-applying-wcag-to-non-web-softwar).
EN 301 549: [PDF of EN 301 549](https://www.etsi.org/deliver/etsi_en/301500_301599/301549/03.02.01_60/en_301549v030201p.pdf), which applies WCAG principles for Name, Role, Value to 9.4.1.2 web, 10.4.1.2 non-web documents, and 11.4.1.2.1 software.

**Environment**:
- Grafana version: 12.2.0
- Data source type & version: Azure Monitor 12.2.0
- User OS & Browser: Chrome or Edge
- Others:

**Comment by alyssajoyner** [comment]
Removing this from the Partner Data sources board as this is not a data source issue 😊

**Comment by v-kavyagupta** [comment]
We verified this issue on the URl:http://localhost:3000/dashboard/new?orgId=1&from=now-7d&to=now&timezone=browser&showCategory=Panel%20options&editview=annotations&editIndex=1
Issue is still repro,  Services label is not programmatically associated with Metrics combo box control.
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/bad2bdb0-fc4d-4b83-8f98-a1b3d2478caf" />
